### PR TITLE
fixing signature of chord.__call__

### LINKED ISSUE #91: Calling a chord object should return an AsyncResult
When you "call" a chord object, it [delegates](https://github.com/celery/celery/blob/master/celery/canvas.py#L1615) to `apply_async`, so the return value of something like
```python
chord([task1.s(), task2.s()])(callback.s())
```
should be `AsyncResult`

I tried adding this:
```python
    def __call__(
        self,
        body: Optional[Signature[Any]] = ...,
        **options: Any,
    ) -> celery.result.AsyncResult[Any]: ...
```

but mypy complains because this disagrees with Signature's typing:
```
typings/celery/canvas.pyi:376:5: error: Signature of "__call__" incompatible with supertype "Signature"  [override]
typings/celery/canvas.pyi:376:5: note:      Superclass:
typings/celery/canvas.pyi:376:5: note:          def __call__(self, *partial_args: Any, **partial_kwargs: Any) -> Any
typings/celery/canvas.pyi:376:5: note:      Subclass:
typings/celery/canvas.pyi:376:5: note:          def __call__(self, body: Optional[Signature[Any]] = ..., **options: Any) -> AsyncResult[Any]
```

is it possible to type this correctly?

### PR DESCRIPTION
**Pull request by galonsky**
Fixes #91

### CODE REVIEW

File: celery-stubs/worker/request.pyi [hunk 1]
```diff
@@ -137,7 +138,7 @@ class Request:
     def reject(self, requeue: bool = ...) -> None: ...
     def info(self, safe: bool = ...) -> _InfoDict: ...
     def humaninfo(self) -> str: ...
-    def chord(self) -> Any | None: ...
+    def chord(self) -> chord_type: ...
```
Comment (galonsky) [inline celery-stubs/worker/request.pyi:141]: actually not sure on this one. should it be optional[chord_type]?

Comment (sbdchd) [reply celery-stubs/worker/request.pyi:141]: Hmm I think looking at the code it should be `Optional[chord_type]` aka `chord_type | None`

https://github.com/celery/celery/blob/018f996fe2f80ed2e69ffd93bd9b35dfa41d1141/celery/worker/request.py#L665

Comment (galonsky) [reply celery-stubs/worker/request.pyi]: thanks, updated

**Comment by sbdchd** [comment]
Looks great, thank you!!

**Pull request merged**